app/src/main/java/dev/andrylat/raqimbek/bankingutils/CardValidator.java

```diff
@@ -0,0 +1,124 @@
+package dev.andrylat.raqimbek.bankingutils;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.stream.IntStream;
+
+public class CardValidator {
+    public CardValidationInfo checkCardNumber(String input) {
+        var errors = new ArrayList<String>();
+        var isValid = false;
+        
+        if (!containsOnlyDigits(input)) {
+            errors.add("Card Number must contain only digits");
+        }
+        
+        for (PaymentSystem paymentSystem : PaymentSystem.values()) {
+            if (!hasValidPrefix(input, paymentSystem)) {
+                errors.add("Payment System can't be determined");
```

Comment:
it means that for EACH payment system the same message will be generated. Are you sure that you want to see the same message 5-10 times when card is invalid?